src/kubernetes.rs

```diff
@@ -0,0 +1,182 @@
+    ///  - similar to command.rs: a bit looser pattern matching on /proc/$pid/cmdline
+    ///  - the following:
+    fn lookup(&self, container_id: &str) -> Result<libc::pid_t> {
+        let (namespace, pod_name, container_name) = try_with!(
+            parse_userinput(container_id),
+            "cannot parse user given container_id"
+        );
+        let containerdid = try_with!(
+            get_containerd_id(namespace, pod_name, container_name),
+            "containerd id lookup failed"
+        );
+        let cgroup = try_with!(find_cgroup(containerdid), "cannot find matching cgroup");
+        let pid = try_with!(
+            get_cgroup_pid(&cgroup),
+            "cannot determine a singular pid owning the cgroup {:?}",
+            cgroup
+        );
+        Ok(pid)
+    }
+
+    fn check_required_tools(&self) -> Result<()> {
+        if cmd::which("kubectl").is_some() {
+            Ok(())
+        } else {
+            bail!("kubectl not found")
+        }
+    }
+}
+
+/// allows the user to prepend the pod name with `custom-namespace/pod-name` to override the
+/// namespace (`default`). By default this will take the first container of the pod. That however
+/// can be overridden by appending it like `namespace/podname/container`.
+pub fn parse_userinput(container_id: &str) -> Result<(&str, &str, Option<&str>)> {
+    let fields = container_id.splitn(3, '/').collect::<Vec<&str>>();
+    if fields.len() == 1 {
+        return Ok((DEFAULT_NAMESPACE, container_id, None));
+    } else if fields.len() == 2 {
+        return Ok((fields[0], fields[1], None));
+    } else if fields.len() == 3 {
+        return Ok((fields[0], fields[1], Some(fields[2])));
+    }
+    unreachable!();
+}
+
+/// find `containerd://hash` id and return hash
+pub fn get_containerd_id(
+    namespace: &str,
+    pod_name: &str,
+    container_name: Option<&str>,
+) -> Result<String> {
+    // kubectl get --raw "/api/v1/namespaces/knative-serving/pods/autoscaler-589958b7b6-l4cb6"
+    // equivalent to `kubectl describe -n knative-serving autoscaler-589958b7b6-l4cb6`
+    let mut url = Url::parse("https://something/api/v1/").unwrap();
+    url.path_segments_mut()
+        .unwrap()
+        .push("namespaces")
+        .push(namespace)
+        .push("pods")
+        .push(pod_name);
+    let url = url.path();
+    let result = try_with!(
+        Command::new("kubectl")
+            .arg("get")
+            .arg("--raw")
+            .arg(url)
+            .output(),
+        "kubctl command cannot be spawned"
+    );
+    if !result.status.success() {
+        let stderr = String::from_utf8_lossy(&result.stderr);
+        bail!(
+            "kubectl get pod request failed (ret code {:?}): {}",
+            result.status.code(),
+            stderr
+        );
+    }
+
+    // parse response. Spec:
+    // https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Pod
+    let json: json::Value = try_with!(
```

Comment:
Serde takes awful long to compile. Do you mind using:

```
$ kubectl get pod -o jsonpath='{range .items[*].status.containerStatuses[*]}{.name}{"\t"}{.containerID}{"\n"}{end}'
runner  containerd://c6d16d9d996e23badc95bc374b86cd8284b9ef76f8650430c58bd28053896234
runner  containerd://eb3ff8ce29b2cca446db2f46b8fbee03dfe032087a8c4df993a39caf137f33ee
```

?

Should be easy to process using split?

Reply:
yes 2c14a76

Reply:
this PR does not add any deps anymore